src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ModifyRolesForUserTest.java

```diff
@@ -253,6 +253,31 @@ public void ac4_delete_role_existing_user_using_pui_user_manager_for_external_20
         //assertThat(!rolesInfo.contains("pui-organisation-manager"));
     }
 
+    @Test
+    public void ac5_modify_role_existing_user_to_organisation_internal_returns_400_if_org_id_invalid() {
+        Map<String, Object> response = professionalApiClient.createOrganisation();
+        String orgIdentifier = (String) response.get("organisationIdentifier");
+        professionalApiClient.updateOrganisation(orgIdentifier, hmctsAdmin);
+
+        IdamOpenIdClient idamOpenIdClient = new IdamOpenIdClient(configProperties);
+        String email = idamOpenIdClient.createUser("pui-organisation-manager");
+        NewUserCreationRequest newUserCreationRequest = professionalApiClient.createNewUserRequest(email);
+        Map<String, Object> newUserResponse = professionalApiClient.addNewUserToAnOrganisation(orgIdentifier, hmctsAdmin, newUserCreationRequest, HttpStatus.CREATED);
+
+        Map<String, Object> searchResponse = professionalApiClient.searchOrganisationUsersByStatusInternal(orgIdentifier, hmctsAdmin, HttpStatus.OK);
+        List<Map> professionalUsersResponses = (List<Map>) searchResponse.get("users");
+        Map professionalUsersResponse = getActiveUser(professionalUsersResponses);
+        String userId = (String) professionalUsersResponse.get("userIdentifier");
+
+        UserProfileUpdatedData userProfileUpdatedData = new UserProfileUpdatedData();
+        RoleName role1 = new RoleName("pui-user-manager");
+        Set<RoleName> roles = new HashSet<>();
+        roles.add(role1);
+        userProfileUpdatedData.setRolesAdd(roles);
+
+        professionalApiClient.modifyUserToExistingUserForPrdAdmin(HttpStatus.INTERNAL_SERVER_ERROR, userProfileUpdatedData, "%7C", userId);

```

Comment:
it should be 400 here

Reply:
As previously discussed, HTTP Status returns 500 due to _RequestRejectedException: The request was rejected because the URL contained a potentially malicious String "%25"_

This error is thrown by Spring before our validation checks the String.